packages/rum/src/domain/record/startFullSnapshots.spec.ts

```diff
@@ -2,51 +2,90 @@ import type { RumConfiguration, ViewCreatedEvent } from '@datadog/browser-rum-co
 import { LifeCycle, LifeCycleEventType } from '@datadog/browser-rum-core'
 import type { TimeStamp } from '@datadog/browser-core'
 import { isIE, noop } from '@datadog/browser-core'
+import { mockExperimentalFeatures, mockRequestIdleCallback } from '@datadog/browser-core/test'
 import type { BrowserRecord } from '../../types'
+import { ExperimentalFeature } from '../../../../core/src/tools/experimentalFeatures'
 import { startFullSnapshots } from './startFullSnapshots'
 import { createElementsScrollPositions } from './elementsScrollPositions'
 import type { ShadowRootsController } from './shadowRootsController'
 
 describe('startFullSnapshots', () => {
   const viewStartClock = { relative: 1, timeStamp: 1 as TimeStamp }
   let lifeCycle: LifeCycle
-  let fullSnapshotCallback: jasmine.Spy<(records: BrowserRecord[]) => void>
+  let fullSnapshotPendingCallback: jasmine.Spy<() => void>
+  let fullSnapshotReadyCallback: jasmine.Spy<(records: BrowserRecord[]) => void>
+  const originalRequestIdleCallback = window.requestIdleCallback
+  const originalCancelIdleCallback = window.cancelIdleCallback
 
   beforeEach(() => {
     if (isIE()) {
       pending('IE not supported')
     }
 
     lifeCycle = new LifeCycle()
-    fullSnapshotCallback = jasmine.createSpy()
+    mockExperimentalFeatures([ExperimentalFeature.ASYNC_FULL_SNAPSHOT])
+    fullSnapshotPendingCallback = jasmine.createSpy('fullSnapshotPendingCallback')
+    fullSnapshotReadyCallback = jasmine.createSpy('fullSnapshotReadyCallback')
+
     startFullSnapshots(
       createElementsScrollPositions(),
       {} as ShadowRootsController,
       lifeCycle,
       {} as RumConfiguration,
       noop,
-      fullSnapshotCallback
+      fullSnapshotPendingCallback,
+      fullSnapshotReadyCallback
     )
   })
 
+  afterEach(() => {
+    window.requestIdleCallback = originalRequestIdleCallback
+    window.cancelIdleCallback = originalCancelIdleCallback
+  })
```

Comment:
**💬 suggestion:** ‏This shouldn't be needed?

Reply:
TODO: remove this afterEach because we don't change requestIdleCallback in this spec file